### DISCUSSION TITLE
RangeCalendar get first selected date

### DISCUSSION THREAD

**Discussion by aaron5670** [mantinedev/mantine#634] [Q&A] [answered]
Hi everyone, 

I really like the Mantine RangeCalendar!

But I have one issue... 
When a user select a date range, I want the first value, before the user selects the second (end) date.
Now when a user selects the first date, the onChange will trigger but returns an array with null like:
```javascript
[null, null]
```

I want to get the first value, before the user selects the second (end) date of the range, like this:

```javascript
<RangeCalendar
  amountOfMonths={2}
  value={value}
  minDate={new Date()}
  
  //Something like this
  onFirstDateChange={(date) => console.log(date)}
  
  onChange={(dates) => {
    //On first date change (first selected date)
    console.log(dates[0]);

    //On second date change (second selected date)
    console.log(dates[1]);

    setValue(dates);
  }}
  fullWidth
/>
```

Could someone help me with this or is there already an existing solution for this that I am overlooking?

Thanks! 😄

**Comment by rtivital** [comment]
Hi Aaron, thanks for kind words! Currently, getting first date is not possible, I'll try to add this feature in one of the next 3.x releases

**Reply by UltraInstinct05** [reply]
@rtivital I could work on this if there are no objections. Should the `onChange` return `[first_date, null]` instead of `[null, null]` on first date selection ?

**Reply by rtivital** [reply]
Yeah, that would be great, thanks! This change should also impact DateRangePicker component: while only one date selected, it should display `12 November – ` value (currently it does not display anything). DateRangePicker should also clear value (call onChange with `[null, null]`) if dropdown was closed before second date was picked.

**Reply by aaron5670** [reply]
> Yeah, that would be great, thanks! This change should also impact DateRangePicker component: while only one date selected, it should display 12 November – value (currently it does not display anything). DateRangePicker should also clear value (call onChange with [null, null]) if dropdown was closed before second date was picked.

Hi @rtivital,

I managed to get the above changes working (#734). 🙂

**Comment by aaron5670** [comment]
Hi @Ultra-Instinct-05 ,

Do you maybe have any status update? 🙂

**Reply by UltraInstinct05** [reply]
Been a bit busy with some stuff at work and some other open source. I'll try to submit a PR this weekend or early next week.

**Reply by UltraInstinct05** [reply]
> if dropdown was closed before second date was picked

@rtivital Question: In the `DateRangePicker`, how do we detect if the popup was closed using <kbd>escape</kbd>, or click outside ?

**Reply by rtivital** [reply]
This behavior is handled in DatePickerBase component, and it should not differ based on the way user closed the dropdown, so you can use the same logic for both ways.

**Reply by aaron5670** [reply]
Hi @Ultra-Instinct-05,

Do you have maybe a new status update for me? 🙂

Please let me know if I can help you! :)

**Comment by aaron5670** [comment]
Hi @Ultra-Instinct-05,

I've managed to get the RangeCalendar working (#734).

But I'm getting stuck with the DateRangePicker component...
Could you look at this if you have time? Or maybe you were already working on this?

Please let me know! 🙂

**Reply by UltraInstinct05** [reply]
Hi @aaron5670 Sorry for the late reply. Had health issues for some time so open source had to take a back burner for a while. I can take a look once I am feeling well (if someone is interested, they can do so as well). Glad you got the PR up for the first date though !

**Reply by aaron5670** [reply]
Hi @Ultra-Instinct-05,

Thank you! I wish you well!

I'm going to look at it again.
Should I not succeed, I will seek help.

**Comment by aaron5670** [comment, marked as answer]
Feature added in #734. 🙂
